hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicyDefault.java

```diff
@@ -393,16 +411,34 @@ private void deleteCheckpoint(Path trashRoot, boolean deleteImmediately)
       }
 
       if (((now - deletionInterval) > time) || deleteImmediately) {
-        if (fs.delete(path, true)) {
-          LOG.info("Deleted trash checkpoint: "+dir);
-        } else {
-          LOG.warn("Couldn't delete checkpoint: " + dir + " Ignoring.");
-        }
+        deleteCheckpoint(path);
       }
     }
+    return counter;
   }
 
-  private long getTimeFromCheckpoint(String name) throws ParseException {
+  /**
+   * Delete a checkpoint
+   * @param path path to delete
+   * @throws IOException IO exception raised in delete call.
+   */
+  protected void deleteCheckpoint(final Path path) throws IOException {
+    String dir = path.toUri().getPath();
+    if (getFileSystem().delete(path, true)) {
+      LOG.info("Deleted trash checkpoint: {}", path);
+    } else {
+      LOG.warn("Couldn't delete checkpoint: {}. Ignoring.", path);
```

Comment:
add a check to see if path exists. if it doesn't exist any more, just log at info rather than warn